products/batch_exports/backend/temporal/destinations/azure_blob_batch_export.py

```diff
@@ -0,0 +1,448 @@
+                ),
+            )
+        except OverBillingLimitError:
+            return
+
+        insert_inputs = AzureBlobInsertInputs(
+            container_name=inputs.container_name,
+            prefix=inputs.prefix,
+            team_id=inputs.team_id,
+            data_interval_start=data_interval_start.isoformat() if not should_backfill_from_beginning else None,
+            data_interval_end=data_interval_end.isoformat(),
+            compression=inputs.compression,
+            exclude_events=inputs.exclude_events,
+            include_events=inputs.include_events,
+            file_format=inputs.file_format,
+            max_file_size_mb=inputs.max_file_size_mb,
+            run_id=run_id,
+            backfill_details=inputs.backfill_details,
+            is_backfill=is_backfill,
+            batch_export_model=inputs.batch_export_model,
+            batch_export_schema=inputs.batch_export_schema,
+            batch_export_id=inputs.batch_export_id,
+            destination_default_fields=azure_blob_default_fields(),
+            integration_id=inputs.integration_id,
+        )
+
+        await execute_batch_export_using_internal_stage(
+            insert_into_azure_blob_activity_from_stage,
+            insert_inputs,
+            interval=inputs.interval,
+        )
+
+
+@activity.defn
+@handle_non_retryable_errors(NON_RETRYABLE_ERROR_TYPES)
+async def insert_into_azure_blob_activity_from_stage(inputs: AzureBlobInsertInputs) -> BatchExportResult:
+    bind_contextvars(
+        team_id=inputs.team_id,
+        destination="AzureBlob",
+        data_interval_start=inputs.data_interval_start,
+        data_interval_end=inputs.data_interval_end,
+    )
+    external_logger = EXTERNAL_LOGGER.bind()
+
+    azure_integration = await _get_azure_blob_integration(inputs.integration_id, inputs.team_id)
+
+    blob_key = get_blob_key(
+        prefix=inputs.prefix,
+        data_interval_start=inputs.data_interval_start,
+        data_interval_end=inputs.data_interval_end,
+        batch_export_model=inputs.batch_export_model,
+        file_format=inputs.file_format,
+        compression=inputs.compression,
+        is_splitting=inputs.max_file_size_mb is not None,
+    )
+
+    external_logger.info(
+        "Batch exporting range %s - %s to Azure Blob: %s",
+        inputs.data_interval_start or "START",
+        inputs.data_interval_end or "END",
+        blob_key,
+    )
+
+    async with Heartbeater():
+        queue = RecordBatchQueue(max_size_bytes=settings.BATCH_EXPORT_AZURE_BLOB_RECORD_BATCH_QUEUE_MAX_SIZE_BYTES)
+        producer = ProducerFromInternalStage()
+
+        assert inputs.batch_export_id is not None
+        producer_task = await producer.start(
+            queue=queue,
+            batch_export_id=inputs.batch_export_id,
+            data_interval_start=inputs.data_interval_start,
+            data_interval_end=inputs.data_interval_end,
+            max_record_batch_size_bytes=1024 * 1024 * 60,  # 60MB
```

Comment:
Matches batch size used by [S3](https://github.com/PostHog/posthog/blob/master/products/batch_exports/backend/temporal/destinations/s3_batch_export.py#L396) and [BigQuery](https://github.com/PostHog/posthog/blob/master/products/batch_exports/backend/temporal/destinations/bigquery_batch_export.py#L963) (60MB).